src/bool_query.py

```diff
@@ -0,0 +1,160 @@
+import tt
+
+from typing import List, Any, Dict
+from enum import Enum
+
+from config import PATH_STOP_WORDS
+from preprocess import tokenize_document, remove_stop_words_from_document, load_stop_words, lemmatize_one_token, get_lemmatizer
+
+class LOGICAL_TOKENS(Enum):
+    AND = "and"
+    OR = "or"
+    NAND = "nand"
+    # NOT = "not"
+
+LOGICAL_TOKENS_VALUES = [x.value for x in LOGICAL_TOKENS]
+
+
+def lemmatize_query(query: str) -> tt.ExpressionTreeNode:
+    tokens = tokenize_document(query)
+    stop_words = load_stop_words(PATH_STOP_WORDS)
+    tokens = remove_stop_words_from_document(tokens, stop_words, LOGICAL_TOKENS_VALUES)
+
+
+    lemmatized_query = []
+    lemmatizer = get_lemmatizer()
+    for tok in tokens:
+        if tok in LOGICAL_TOKENS_VALUES:
+            lemmatized_query.append(tok)
+        else:
+            lemmatized_query.append(lemmatize_one_token(tok, lemmatizer))
```

Comment:
It is preferable to apply pos tagging to entire phrases or chunks of phrases. More info in the preprocess review.

Reply:
ok, changed